src/components/operations/operation-details/react/runtime/OperationDetails.tsx

```diff
@@ -0,0 +1,402 @@
+    Subtitle2,
+    Table,
+    TableBody,
+    TableCell,
+    TableHeader,
+    TableHeaderCell,
+    TableRow,
+    Tooltip 
+} from "@fluentui/react-components";
+import { Copy16Regular } from "@fluentui/react-icons";
+import { RouteHelper } from "../../../../../routing/routeHelper";
+import { ApiService } from "../../../../../services/apiService";
+import { Api } from "../../../../../models/api";
+import { Operation } from "../../../../../models/operation";
+import { Tag } from "../../../../../models/tag";
+import { Request } from "../../../../../models/request";
+import { Response } from "../../../../../models/response";
+import {
+    TypeDefinition,
+    TypeDefinitionProperty,
+    TypeDefinitionPropertyTypeArrayOfPrimitive,
+    TypeDefinitionPropertyTypeArrayOfReference,
+    TypeDefinitionPropertyTypeCombination,
+    TypeDefinitionPropertyTypePrimitive,
+    TypeDefinitionPropertyTypeReference
+} from "../../../../../models/typeDefinition";
+import { MarkdownProcessor } from "../../../../utils/react/MarkdownProcessor";
+import { OperationDetailsRuntimeProps } from "./OperationDetailsRuntime";
+import { OperationRepresentation, TSchemaView } from "./OperationRepresentation";
+import { TypeDefinitionInList } from "./TypeDefinitions";
+import { OperationDetailsTable, getRequestUrl, scrollToOperation } from "./utils";
+
+export const OperationDetails = ({
+    apiName,
+    operationName,
+    apiService,
+    routeHelper,
+    enableConsole,
+    useCorsProxy,
+    includeAllHostnames,
+    enableScrollTo,
+    showExamples,
+    defaultSchemaView
+}: OperationDetailsRuntimeProps & { apiName: string, operationName: string, apiService: ApiService, routeHelper: RouteHelper }) => {
+    const [working, setWorking] = useState<boolean>(false);
+    const [api, setApi] = useState<Api>(null);
+    const [operation, setOperation] = useState<Operation>(null);
+    const [tags, setTags] = useState<Tag[]>([]);
+    const [request, setRequest] = useState<Request>(null);
+    const [responses, setResponses] = useState<Response[]>(null);
+    const [hostnames, setHostnames] = useState<string[]>([]);
+    const [definitions, setDefinitions] = useState<TypeDefinition[]>([]);
+    const [requestUrl, setRequestUrl] = useState<string>(null);
+    const [isCopied, setIsCopied] = useState<boolean>(false);
+
+    useEffect(() => {
+        if (!apiName) return;
+        
+        setWorking(true);
+        loadApi().then(loadedApi => setApi(loadedApi));
+        loadGatewayInfo().then(hostnames => {
+            hostnames?.length > 0 && setHostnames(hostnames);
+        });
+        loadOperation().then(loadedValues => {
+            setOperation(loadedValues.operation);
+            setTags(loadedValues.tags);
+            setDefinitions(loadedValues.definitions);
+            setRequest(loadedValues.operation?.request);
+            setResponses(loadedValues.operation?.getMeaningfulResponses());
+        }).finally(() => {
+            setWorking(false);
```

Comment:
I don't think this works as expected? what if loadApi takes longer than loadOperation?

maybe use Promise.all?

Also maybe at least some basic error handling?